uint32 state
uint32 cmd
uint32 target_state
uint32 time_steps
uint32 new_state
float32 error
